hangman.py

```diff
@@ -0,0 +1,240 @@
+                print(get_guessed_word(secret_word, letters_guessed))
+            else:
+                letters_guessed, guesses_point = letter_in_word(secret_word, letter, letters_guessed, guesses_point, letters_guessed)
+        if is_word_guessed(secret_word, letters_guessed) == 0 or guesses_point <= 0:
+            break
+
+    #output result of game
+    if is_word_guessed(secret_word, letters_guessed) == 0:
+        final_points = guesses_point * len(set(secret_word))
+        print("------")
+        print("Congratulations, you won!")
+        print("Word:", secret_word)
+        print("Your total score for this game is:", final_points)
+    elif is_word_guessed(secret_word, letters_guessed) != 0 and guesses_point <= 0:
+        print("------")
+        print("Sorry, you ran out of guessed. The word was", secret_word)
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise:
+    '''
+
+    if len(my_word) == len(other_word):
+        for i in range(len(my_word)):
+            if (my_word[i] != '_' and my_word[i] != other_word[i]) or (my_word[i] == "_" and other_word[i] in my_word):
+                return False
+        return True
+    else:
+        return False
+
+
+def show_possible_matches(my_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    returns: nothing, but should print out every word in wordlist that matches my_word
+             Keep in mind that in hangman when a letter is guessed, all the positions
+             at which that letter occurs in the secret word are revealed.
+             Therefore, the hidden letter(_ ) cannot be one of the letters in the word
+             that has already been revealed.
+    '''
+
+    list_with_words = []
+    my_word = my_word.replace(" ", "")
+
+    for word in list(wordlist):
+        if match_with_gaps(my_word, word) is True:
+            list_with_words.append(word)
+
+    print("Possible word matches are:", " ".join(list_with_words))
+
+
+def request_type():
+    print("Welcome to the game Hangman!")
+    print("Do you want play hangman (classic) or hangman with hints?")
+
+    while True:
+        request = input("if you want to play hangman (classic) input 'classic', play hangman with hints input 'hints':").replace(" ", "")
+        if request == "classic" or request == "hints":
+            start_hangman(request)
+            break
+        else:
+            print("Oops! Your input is not correct")
+
+
+def start_hangman(request):
+    if request == "classic":
+        request = True
+    elif request == "hints":
+        request = False
```

Comment:
I believe you can move that code into `request_type` function, it would be more logical IMO